pep-0681.rst

```diff
@@ -0,0 +1,709 @@
+We considered supporting ``auto_attribs`` and a corresponding
+``auto_attribs_default`` parameter. We decided against this because it
+is specific to attrs and appears to be a legacy behavior. Instead of
+supporting this in the new standard, we recommend that the maintainers
+of attrs move away from the legacy semantics and adopt
+``auto_attribs`` behaviors by default.
+
+Django does not support declaring fields using type annotations only,
+so users of this mechanism would need to know that they should always
+supply assigned values.
+
+``cmp`` parameter
+-----------------
+The attrs library supports a bool parameter ``cmp`` that is equivalent
+to setting both ``eq`` and ``order`` to True. This is not supported in
+this proposal. Attrs users should use the dataclass-standard parameter
+names.
+
+``kw_only`` field descriptor parameter
+--------------------------------------
+The attrs library supports a ``kw_only`` parameter for individual
+fields. This is not currently supported in this spec, but it could be
+added in the future if there was sufficient demand.
+
+Automatic field name aliasing
+-----------------------------
+The attrs library performs automatic aliasing of field names that
+start with a single underscore. This proposal omits this behavior.
+
+Alternate field ordering algorithms
+-----------------------------------
+The attrs library currently supports two approaches to ordering the
+fields within a class:
+
+* Dataclass order: The same ordering used by dataclasses. This is the
+  default behavior of the older APIs (ex. attr.s).
+* Method Resolution Order (MRO): This is the default behavior of the
+  newer APIs (ex. define, mutable, frozen). Older APIs (ex. attr.s)
+  can opt into this behavior by specifying ``collect_by_mro=True``.
+
+The resulting field orderings can differ in certain diamond-shaped
+multiple inheritance scenarios.
+
+This proposal does not support any field ordering other than that used
+by dataclass.
+
+Fields redeclared in subclasses
+-------------------------------
+The attrs library differs from stdlib dataclasses in how it
+handles inherited fields that are redeclared in subclasses. The
+dataclass specification preserves the original order, but attrs
+defines a new order based on subclasses. Users of attrs who rely on
+this ordering will not see the expected order of parameters in the
+synthesized ``__init__`` method.
+
+Django primary and foreign keys
+-------------------------------
+Django applies additional logic for primary keys and
+foreign keys. For example, it automatically adds an "id" field (and
+``__init__`` parameter) if there is no field designated as a primary
+key. This additional logic is not accommodated with this proposal, so
+users of Django would need to explicitly declare the id field.
+
+This limitation may make it impractical to use the
+``dataclass_transform`` mechanism with Django.
+
+Open Issues
+===========
+
+``converter`` field descriptor parameter
```

Comment:
Doesn't pydantic have this too—in fact, isn't it one of its central features? Might be worth mentioning here, especially given its greater modern popularity relative to attrs

Reply:
Could be. Unfortunately I'm not a pydantic expert. I scanned through their docs and didn't see anything similar.

@samuelcolvin, does pydantic have a feature similar to attrs' converter field descriptor parameter?